### PR TITLE
[tests-only][full-ci]Add api tests to assign role using graph api

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This scenario adds scenario assign a role to a user using the graph API and check if its correctly listed by the graph API and and by the settings API

## Related Issue
- https://github.com/owncloud/ocis/issues/5565#

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by amrita-shrestha** [comment]
blocked until this issue confirm as a bug from the developer side https://github.com/owncloud/ocis/issues/5583?

File: tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature
Comment (SagarGi) [inline tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature:90]: how about moving this to grapAPI suite ? this feature seems like of `setting api`

File: tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature
Comment (SagarGi) [inline tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature:81]:
```suggestion
    And the administrator has given "Alice" the role "<userRole>" using the setting api
```
is this something like this as your scenario description says.

File: tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature
Comment (SagarGi) [inline tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature:84]:
```suggestion
    And the Graph API response should have the role "<userRole>"
```

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/GraphContext.php:1514]: i see this `applicationId` used only to this function. is it used any where esle that you made it a class variable?

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/GraphContext.php:1528]:
```suggestion
			throw new Error('Could not assign role ' . $role);
```

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/GraphContext.php:1565]:
```suggestion
			throw new Error('App role should be ' . $role . ' but found ' . $response['appRoleId']);
```

File: tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature
Comment (amrita-shrestha) [reply tests/acceptance/features/apiAccountsHashDifficulty/assignRole.feature:84]: its setting api

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (amrita-shrestha) [reply tests/acceptance/features/bootstrap/GraphContext.php:1514]: it will be used on my follow-up PR 😸 . I forget to do clean up

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (grgprarup) [inline tests/acceptance/features/apiGraph/assignRole.feature:9]: How about creating a user using the graph API?

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (grgprarup) [inline tests/acceptance/features/apiGraph/assignRole.feature:22]:
```suggestion
  Scenario Outline: assign role to the user with graph API and list role with settings API
```
Isn't it like this?

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (SwikritiT) [reply tests/acceptance/features/apiGraph/assignRole.feature:9]: isn't the user created through this step is created with graph api?

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/GraphContext.php:1575]:
```suggestion
	 * @When /^the administrator gives  "([^"]*)" the role "([^"]*)" using the Graph API$/
```

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/GraphContext.php:1585]:
```suggestion
	public function theAdministratorGivesTheRoleUsingTheGraphApi(string $user, string $role): void {
```

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/GraphContext.php:1588]:
```suggestion
		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id') ?? user;
```
Can be done like this as well?

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/GraphContext.php:1623]:
```suggestion
		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id') ?? $user;
```
same here

File: tests/acceptance/features/bootstrap/RoleAssignmentContext.php
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/RoleAssignmentContext.php:262]: setting or graph or are they the same?

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/assignRole.feature:5]: how about adding scenarios related to `a normal user tries to get roles, assign roles and get list of assignments` if missing.

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/assignRole.feature:24]:
```suggestion
    And the administrator has assigned user "Alice" the role "<userRole>" using the Graph API
```

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (SagarGi) [inline tests/acceptance/features/apiGraph/assignRole.feature:10]:
```suggestion
    And the administrator has assigned "Alice" the role "<userRole>" using the Graph API
```

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (amrita-shrestha) [reply tests/acceptance/features/apiGraph/assignRole.feature]: please look issue mentioned in this PR

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (amrita-shrestha) [reply tests/acceptance/features/apiGraph/assignRole.feature:5]: no not needed such scenario as confirmed by kulman

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (saw-jan) [reply tests/acceptance/features/apiGraph/assignRole.feature:9]: The step creates the user using the Graph API

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/assignRole.feature:26]: `GET` returns `201 Created` status code?

File: tests/acceptance/features/bootstrap/RoleAssignmentContext.php
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/RoleAssignmentContext.php:131]:
```suggestion
                $userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
		$this->setRoleToUser($admin, $userId, $roleId);
```

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (grgprarup) [reply tests/acceptance/features/apiGraph/assignRole.feature:9]: @SwikritiT Yeah, this step works for both provisioning and graph, AFAIK we are using graph API only for users-related CRUD in Ocis. This step should be cleaned up in the near future. Also, we have a dedicated graph API step.
Decision is yours @amrita-shrestha

Comment (SwikritiT) [reply tests/acceptance/features/apiGraph/assignRole.feature:9]: Yes, the step needs to be refactored but that shouldn't be in this PR. This should actually  be a part of https://github.com/owncloud/ocis/issues/5337 or we can create separate issue

**Review by SagarGi — COMMENTED** [review]
LGTM :+1: if CI is green

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/assignRole.feature:22]:
```suggestion
  @issue-5032
  Scenario Outline: assign role to the user with graph api and list role with setting api
```

File: tests/acceptance/features/bootstrap/RoleAssignmentContext.php
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/RoleAssignmentContext.php:270]: use this if possible: `$this->featureContext->getJsonDecodedResponse()`
```suggestion
		$assignmentRoleId = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())["assignments"][0]["roleId"];
```

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (saw-jan) [inline tests/acceptance/features/apiGraph/assignRole.feature:10]:
```suggestion
    And the administrator has assigned the role "<userRole>" to user "Alice" using the Graph API
```

File: tests/acceptance/features/bootstrap/GraphContext.php
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/GraphContext.php:1577]:
```suggestion
	 * @param string $role
	 * @param string $user
```

File: tests/acceptance/features/apiGraph/assignRole.feature
Comment (SwikritiT) [inline tests/acceptance/features/apiGraph/assignRole.feature:5]:
```suggestion
  As an admin
  I want to assign roles to users
  So that I can provide them with authority 
```

**Review by SagarGi — COMMENTED** [review]
others LGTM :+1:

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by individual-it — APPROVED** [review]

**Pull request merged**
